qutip/lattice.py

```diff
@@ -0,0 +1,785 @@
+    position operators, Hamiltonian in the positio represention etc.
+
+    Parameters
+    ----------
+    num_cell : int
+        The number of cells in the crystal.
+    boundary : str
+        Specification of the type of boundary the crystal is defined with.
+    cell_num_site : int
+        The number of sites in the unit cell.
+    cell_site_dof : list of int/ int
+        The tensor structure  of the degrees of freedom at each site of a unit cell.        
+    cell_Hamiltonian :  Qobj        
+        The Hamiltonian of the unit cell.
+    inter_hop : Qobj / list of Qobj
+        The coupling between the unit cell at i and at (i+unit vector)
+                
+    Attributes
+    ----------
+    num_cell : int
+        The number of unit cells in the crystal.
+    cell_num_site : int
+        The nuber of sites in a unit cell.
+    length_for_site : int
+        The length of the dimension per site of a unit cell.
+    cell_tensor_config : list of int
+        The tensor structure of the cell in the form 
+        [cell_num_site,cell_site_dof[:][0] ]
+    lattice_tensor_config : list of int
+        The tensor structure of the crystal in the 
+        form [num_cell,cell_num_site,cell_site_dof[:][0]]
+    length_of_unit_cell : int
+        The length of the dimension for a unit cell.
+    PBCx : int
+        1 indicates "periodic" and 0 indicates "hardwall" boundary condition
+    inter_vec_list : list of list
+        The list of list of coefficients of inter unitcell vectors' components
+        along Cartesian uit vectors.
+    lattice_vectors_list : list of list
+        The list of list of coefficients of lattice basis vectors' components 
+        along Cartesian unit vectors.
+    H_intra : Qobj
+        The Qobj storing the Hamiltonian of the unnit cell.
+    H_inter_list : list of Qobj/ Qobj
+        The list of coupling terms between unit cells of the lattice.           
+    is_consistent : bool
+        Indicates the consistency/correctness of all the attributes together
+        in the unit Lattice1d.
+        
+    Methods
+    -------
+    Hamiltonian()
+        Hamiltonian of the crystal.
+    basis()
+        basis with the particle localized at a certain cell,site with degree of
+        freedom.
+    distribute_operator()
+        Distributes an input operator over all the cells.
+    x()
+        Position operator for the crystal.
+    operator_at_cells()
+        Distributes an input operator over user specified cells .
+    plot_dispersion()
+        Plots dispersion relation of the crystal.
+    get_dispersion()
+        Returns the dispersion relation of the crystal.      
+    """    
+    def __init__(self, num_cell=10, boundary = "periodic", cell_num_site = 1, cell_site_dof = [1], cell_Hamiltonian = Qobj([[ ]]) , inter_hop = Qobj([[ ]])   ):
+
+        self.num_cell = num_cell        
+        self.cell_num_site = cell_num_site
+
+        if ( type(cell_num_site) != int or cell_num_site < 0 ):
```

Comment:
Instead of `type(cell_num_site) != int` use `not isinstance(cell_num_site, int)`